phac-nml/irida-uploader
# Delete Sequencing Run script

## What does this PR do and why?
This adds a script `scripts/delete_sequencing_run.sh` that accepts a config file and a run directory, and deletes the associated sequencing run from IRIDA.
This is needed because sometimes high throughput users upload to the wrong project, and the data needs to be quickly removed and reuploaded.

## Screenshots or screen recordings
![image](https://github.com/phac-nml/irida-uploader/assets/16961365/635b69aa-72b1-4e48-bc85-0329b14d5414)

## How to set up and validate locally

1. Run IRIDA
2. Create create a sequencing client
3. Upload a sequencing run to IRIDA
a. Pull this branch
b. `make gui`
c. `./scripts/gui.sh`
d. open Configuration Settings and enter client/user/url details
e. Upload some data, the example data in `examples/` will work for this
f. Verify on IRIDA that the data exists in the project
e. Verify the Sequencing Run exists in the Admin panel
4. run the new script without any arguments to see usage message `./scripts/delete_sequencing_run.sh`
5. run with --version to see version info `./scripts/delete_sequencing_run.sh --version`
6. run with --help to see help info `./scripts/delete_sequencing_run.sh --help`
7. run with config file and directory specified to start script
a. default config file directory is `-c ~/.config/irida-uploader/config.conf`
b. directory should be the directory you uploaded, ex: `-d examples/miseq_run
8. Verify that the displayed files match what was uploaded
9. Verify that you are prompted with run id to confirm deletion
10. enter run id to delete
11. Verify on IRIDA that the data was deleted from the project
12. Verify on the IRIDA Admin panel that the sequencing run was deleted
13. Run script again, but this time verifying that the run could not be found as it was deleted
14. Upload more data, and run script again, but enter a wrong id and verify that run does not get deleted
15. Manually edit the status file `examples/miseq_run/irida_uploader_status.info`, changing `"Upload Status": "complete"` to  `"Upload Status": "partial"`
16. Run script again, and verify that the script give a message about non complete run, and exits

## PR acceptance checklist
Things for the developer to confirm they've done before the PR should be accepted:

* [x] CHANGELOG.md updated with information for new change.
